File: maximum_average_subarray_1/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    double findMaxAverage(vector<int> &nums, int k)
    {
        int i = 0, j = k;
        double s = accumulate(nums.begin(), nums.begin() + k, 0.0);
        double m = s;
        while (j < nums.size())
        {
            s = s + nums[j] - nums[i];
            m = max(m, s);
            i++;
            j++;
        }
        return m / k;
    }
};

File: maximum_average_subarray_1/solution.js
let solution = function (nums, k) {
    let i = 0, j = k, s = 0;
    for (let i = 0; i < k; i++)
        s += nums[i]
    let m = s;
    while (j < nums.size) {
        s = s - nums[i] + nums[j]
        m = Math.max(s, m)
        i += 1
        j += 1
    }
    return m / k
}